var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -133,15 +138,29 @@ def configure_args(self):
 
         return args
 
-    def installcheck(self):
-        fi_info = Executable(self.prefix.bin.fi_info)
-        fi_info()
+    def install(self, spec, prefix):
+        # Call main install method
+        super(Libfabric, self).install(spec, prefix)
 
-        # Build and run more extensive tests
+        # Build and install fabtests, if available
+        if not os.path.isdir('fabtests'):
+            return
         with working_dir('fabtests'):
             configure = Executable('./configure')
             configure('--prefix={0}'.format(self.prefix),
                       '--with-libfabric={0}'.format(self.prefix))
             make()
             make('install')
+
+    def installcheck(self):
+        fi_info = Executable(self.prefix.bin.fi_info)
+        fi_info()
+
+        # Run fabtests test suite if available
+        if not os.path.isdir('fabtests'):
+            return
+        if self.spec.satisfies('@1.8.0,1.9.0'):
+            # make test seems broken.
+            return
+        with working_dir('fabtests'):
             make('test')
```

Comment:
Maybe run these tests after build instead of after install

Reply:
The current order is like this:

1. configure/build/install libfabric.
2. configure/build fabtests (inside `install()`)
3. install fabtests (inside `install()`)
4. run tests (inside `installcheck()`)

We could try to exchange (3) and (4), but why?